sdk/core/Azure.Core.TestFramework/src/RecordedTestSanitizer.cs

```diff
@@ -33,6 +36,11 @@ public virtual void SanitizeHeaders(IDictionary<string, string[]> headers)
 
         public virtual string SanitizeTextBody(string contentType, string body)
         {
+            foreach (var jsonPath in JsonPathSanitizers)
```

Comment:
We should have a fast-path that skips parsing JSON when there are not paths to sanitize.

Reply:
Could you expand more on what you mean? 
In this case, if `JsonPathSanitizers` doesn't have data on it, the `foreach` won't execute and we will return the body

Reply:
When you implement https://github.com/Azure/azure-sdk-for-net/pull/12707#discussion_r438580472 the loop would be inversed - parsing first, looping next.

Would be nice to skip parsing if there is nothing in the collection.